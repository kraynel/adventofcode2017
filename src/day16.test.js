const { letsDance, letsDanceWithInput } = require("./day16");
const fs = require("fs");

describe("letsDance", () => {
  it("should dance", () => {
    expect(letsDance("s1,x3/4,pe/b", 5)).toBe("baedc");
  });

  it("should get the value", done => {
    fs.readFile("inputs/day16.txt", "utf8", (err, data) => {
      expect(letsDance(data)).toMatchSnapshot();
      done();
    });
  });

  it("should get the value", done => {
    fs.readFile("inputs/day16.txt", "utf8", (err, data) => {
      let val = letsDanceWithInput(data, "abcdefghijklmnop");
      let cycle = 1;
      while (val !== "abcdefghijklmnop") {
        val = letsDanceWithInput(data, val);
        cycle++;
      }

      const toLoop = 1e9 % cycle;

      val = letsDanceWithInput(data, "abcdefghijklmnop");
      for (let i = 0; i < toLoop - 1; i++) val = letsDanceWithInput(data, val);

      expect(val).toMatchSnapshot();
      done();
    });
  });
});
