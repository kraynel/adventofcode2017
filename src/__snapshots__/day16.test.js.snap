// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`letsDance should get the value 1`] = `"glnacbhedpfjkiom"`;

exports[`letsDance should get the value 2`] = `"fmpanloehgkdcbji"`;
